test/suites/snapshots.sh

```diff
@@ -202,8 +203,10 @@ snap_restore() {
   restore_and_compare_fs snap1
 
   # Check that instances UUID remain the same before and after snapshoting
+  # But generation ID should be updated
   newUUID=$(lxc config get bar volatile.uuid)
-  if [ "${initialUUID}" != "${newUUID}" ]; then
+  newGenerationID=$(lxc config get bar volatile.uuid.generation)
+  if [ "${initialGenerationID}" = "${newGenerationID}" ] || [ "${initialUUID}" != "${newUUID}" ]; then
```

Comment:
You're using "friendly" errors here, but you've not updated the error message in response to the change in logic.
You should probably have this as a separate check with a separate error message for generation ID.